src/models/transactions/offerCreate.ts

```diff
@@ -0,0 +1,47 @@
+import { ValidationError } from "../../common/errors";
+import { Amount, IssuedCurrencyAmount } from "../common";
+import { CommonFields, verifyCommonFields } from "./common";
+
+export interface OfferCreate extends CommonFields {
+    TransactionType: "OfferCreate";
+    Expiration?: number;
+    OfferSequence?: number;
+    TakerGets: Amount;
+    TakerPays: Amount;
+}
+
+/**
+ * Verify the form and type of an OfferCreate at runtime.
+ * 
+ * @param tx - An OfferCreate Transaction
+ * @returns - Void.
+ * @throws - When the OfferCreate is Malformed.
+ */
+ export function verifyOfferCreate(tx: OfferCreate): void {
+    verifyCommonFields(tx)
+
+    if (tx.TakerGets === undefined)
+        throw new ValidationError("OfferCreate: missing field TakerGets")
+
+    if (tx.TakerPays === undefined)
+        throw new ValidationError("OfferCreate: missing field TakerPays")
+
+    const isIssuedCurrency = (obj: IssuedCurrencyAmount): boolean => {
+        return Object.keys(obj).length === 3 
```

Comment:
These kinds of runtime checks are ok, but not a big deal. I believe these are the mistakes that rippled would immediately throw out as invalid (probably a `tem`) so we don't really need our library to do this, IMHO

Reply:
These kinds of simple errors are super easy to catch and can save someone a few milliseconds.

Reply:
I'm neutral on this. Personally I don't think it's worth the extra code, but if you like it, then it is fine to include. 👍